catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -155,7 +164,9 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
                 List<Metacard> createdMetacards = createResponse.getCreatedMetacards();
 
                 for (Metacard metacard : createdMetacards) {
-                    fileIdMap.store(key, metacard.getId());
+                    if (safeKey != null) {
```

Comment:
❓  should `if (safeKey != null)` wrap the entire for loop? if `safeKey` is null then it looks like it just logs the debug statement for each metacard.